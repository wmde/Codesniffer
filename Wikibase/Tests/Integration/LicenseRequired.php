<?php

// phpcs:property Wikibase\Sniffs\Commenting\ClassLevelDocumentationSniff::$license = MIT
// phpcs:disable Generic.Files.OneObjectStructurePerFile

class Undocumented {
}

/**
 * @todo Add license.
 */
class Unlicensed {
}

/**
 * @license LGPL-3.0-or-later with unwanted extra text
 */
class WrongLicense {
}

/**
 * @license LGPL-3.0-or-later
 * @licence MIT
 */
class MultiLicensed {
}

/**
 * @inheritDoc
 */class MisplacedEmptyComment {
}

/*
 * @license LGPL-3.0-or-later
 */
class NonPhpDocComment {
}

namespace Nested {

	abstract class Undocumented {
	}

	/**
	 * @inheritDoc
	 */
	class EmptyComment {
	}

	/**
	 * Multiple sniffs detect this, but no other can fix it.
	 * @license
	 */
	class EmptyLicense {
	}

}
