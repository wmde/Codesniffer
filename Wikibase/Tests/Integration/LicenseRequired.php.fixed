<?php

// phpcs:property Wikibase\Sniffs\Commenting\ClassLevelDocumentationSniff::$license = MIT
// phpcs:disable Generic.Files.OneObjectStructurePerFile

/**
 * @license MIT
 */
class Undocumented {
}

/**
 * @todo Add license.
 * @license MIT
 */
class Unlicensed {
}

/**
 * @license MIT
 */
class WrongLicense {
}

/**
 * @license LGPL-3.0-or-later
 * @license MIT
 */
class MultiLicensed {
}

/**
 * @inheritDoc
 * @license MIT
 */
class MisplacedEmptyComment {
}

/**
 * @license MIT
 */
class NonPhpDocComment {
}

namespace Nested {

	/**
	 * @license MIT
	 */
	abstract class Undocumented {
	}

	/**
	 * @inheritDoc
	 * @license MIT
	 */
	class EmptyComment {
	}

	/**
	 * Multiple sniffs detect this, but no other can fix it.
	 * @license MIT
	 */
	class EmptyLicense {
	}

}
