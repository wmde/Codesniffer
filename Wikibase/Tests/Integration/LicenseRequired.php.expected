  6 | ERROR   | [x] @license missing
  9 | ERROR   | [x] @license missing
 16 | ERROR   | [x] Wrong @license
 23 | ERROR   | [x] Found @licence instead of @license
 28 | ERROR   | [x] @license missing
 30 | WARNING | [x] No newline after class level documentation
 35 | ERROR   | [x] Regular comment found instead of class level documentation
 41 | ERROR   | [x] @license missing
 44 | ERROR   | [x] @license missing
 52 | ERROR   | [x] @license not followed by a license
